gems/aws-sdk-core/lib/aws-sdk-core/assume_role_credentials.rb

```diff
@@ -17,6 +17,11 @@ module Aws
   #
   # If you omit `:client` option, a new {STS::Client} object will be
   # constructed.
+  #
+  # The AssumeRoleCredentials also provides a `before_refresh` callback
```

Comment:
Misleading doc

Reply:
Should this be that RefreshingCredentials provides the callback? Will each credential provider have a comment like this?

Reply:
The use of RefreshingCredentials is, to me, a behind the scenes implementation detail.  Someone reading the documentation here shouldn't care how its done.

I dont think each refreshing credential provider should have a comment until we know what use cases they may want to solve with them - the documentation is useful here I think because it gives an idea what you would actually use this for.